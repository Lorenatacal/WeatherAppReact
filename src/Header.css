.Header {
    text-align: center;
  }
  
  .Header-logo {
    animation: App-logo-spin infinite 30s linear;
    height: 90px;
  }
  
  .App-header {
    background-image: linear-gradient(white, skyblue);
    height: 150px;
    padding: 20px;
    color: white;
    margin-left: 25px;
    margin-right: 25px;
    margin-top: 25px;
    border: 1px solid lightgrey;
    border-radius: 20px;
  }
  
  .App-title {
    font-size: 2.8rem;
    font-family: 'Times New Roman', Times, serif;
    text-shadow: -1px 0 rgb(19, 18, 18), 0 1px darkgrey, 1px 0 rgb(19, 18, 18), 0 -1px rgb(19, 18, 18);
  }
  
  @keyframes App-logo-spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
  }