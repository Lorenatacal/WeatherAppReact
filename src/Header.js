import React, { Component } from 'react';
import './Header.css';

class Header extends Component {
    render() {
      return (
        <div className="Header">
          <header className="App-header">
            <img src= "https://img.clipartxtras.com/96da9eda759c08834fa44d63ab4098dd_weather-png-images-transparent-free-download-pngmartcom-weather-clipart-png_512-512.png" className="Header-logo" alt="logo" />
            <h1 className="App-title">{this.props.title}</h1>
          </header>
        </div>
      );
    }
  }
  
  export default Header;