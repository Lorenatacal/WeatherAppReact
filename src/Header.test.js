import React, { Component } from 'react';
import Header from './Header';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test("Header should render correctly", () => {
    const wrapper = Enzyme.shallow(<Header />); //create a wrapper with the rendered component
    expect(wrapper).toMatchSnapshot();
});
