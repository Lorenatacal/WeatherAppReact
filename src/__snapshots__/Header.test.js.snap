// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <header
        className="App-header"
>
        <img
                alt="logo"
                className="Header-logo"
                src="https://img.clipartxtras.com/96da9eda759c08834fa44d63ab4098dd_weather-png-images-transparent-free-download-pngmartcom-weather-clipart-png_512-512.png"
        />
        <h1
                className="App-title"
        />
</header>,
      "className": "Header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="Header-logo"
            src="https://img.clipartxtras.com/96da9eda759c08834fa44d63ab4098dd_weather-png-images-transparent-free-download-pngmartcom-weather-clipart-png_512-512.png"
/>,
          <h1
            className="App-title"
/>,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "Header-logo",
            "src": "https://img.clipartxtras.com/96da9eda759c08834fa44d63ab4098dd_weather-png-images-transparent-free-download-pngmartcom-weather-clipart-png_512-512.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "App-title",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
      ],
      "type": "header",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <header
          className="App-header"
>
          <img
                    alt="logo"
                    className="Header-logo"
                    src="https://img.clipartxtras.com/96da9eda759c08834fa44d63ab4098dd_weather-png-images-transparent-free-download-pngmartcom-weather-clipart-png_512-512.png"
          />
          <h1
                    className="App-title"
          />
</header>,
        "className": "Header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="Header-logo"
              src="https://img.clipartxtras.com/96da9eda759c08834fa44d63ab4098dd_weather-png-images-transparent-free-download-pngmartcom-weather-clipart-png_512-512.png"
/>,
            <h1
              className="App-title"
/>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "Header-logo",
              "src": "https://img.clipartxtras.com/96da9eda759c08834fa44d63ab4098dd_weather-png-images-transparent-free-download-pngmartcom-weather-clipart-png_512-512.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "App-title",
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
        ],
        "type": "header",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
